src/main/java/io/temporal/activity/ActivityOptions.java

```diff
@@ -173,13 +177,19 @@ public ActivityOptions build() {
     }
 
     public ActivityOptions validateAndBuildWithDefaults() {
+      if (retryOptions != null) {
+        // retryOptions being null means we will explicitly fallback to the default
+        // activity retry behavior as specified on the server
+        retryOptions = RetryOptions.newBuilder(retryOptions).validateBuildWithDefaults();
```

Comment:
I think we should not add any defaults to the retry options and rely on the service for these.